core/rule_manager.py

```diff
@@ -80,16 +80,17 @@ def list_rules(self, user_id: str) -> list:
         visited_rule_ids = []
         for rule in rules:
             if rule[0] not in visited_rule_ids:
-                res.append({
-                    'id': rule[0],
-                    'key': rule[1],
-                    'name': rule[2],
-                    'pattern': rule[3],
-                    'constraints': rule[4],
-                    'rewrite': rule[5],
-                    'actions': rule[6],
-                    'enabled_apps': rule_applications[rule[0]]
-                })
+                if app_id == None or (len(rule_applications[rule[0]]) and int(app_id) == rule_applications[rule[0]][0]["app_id"]):
```

Comment:
This is not right. `rule_applications[rule[0]]` is a list of applications that are enabled for this rule. Your current way only checks the first enabled application, if its id is the given `app_id`, this rule will be returned, otherwise this rule will not be returned. What if the `app_id` is the second enabled application for this rule?

Reply:
Got it! I did not touch this part when implementing the Applications tab but I just fixed it to let `app_id` match with all the applications enabled under this rule.